lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_object.rb

```diff
@@ -561,10 +565,10 @@ def self.convert_value_based_on_datatype(value, datatype)
       return convert_boolean_value(value)                    if datatype == 'boolean'
       return true                                            if datatype == 'TrueClass'
       return false                                           if datatype == 'FalseClass'
-      return Time.parse(value)                               if datatype == 'time' || datatype == 'Time'
-      return value.to_sym                                    if datatype == 'symbol' || datatype == 'Symbol'
+      return Time.parse(value).getlocal                      if %w[time Time].include?(datatype)
```

Comment:
Been thinking about these comparisons and want to suggest using `casecmp?` include?  See https://ruby-doc.org/core-2.4.0/String.html

For example:
`'time'.casecmp?(datatype)`

Many of the comparisons here could use that.